### PR TITLE
Updated Integration test docs with missing annotations

### LINKED ISSUE #7936: Add Docs For Missing Integration Test Annotations
# Topic clarification/correction request

## Topic Link

https://devdocs.magento.com/guides/v2.4/test/integration/annotations.html

## Description

The docs do not list all possible integration test annotations. The following annotations are missing:

* `@magentoAdminConfigFixture`
* `@magentoDataFixtureBeforeTransaction`
* `@magentoIndexerDimensionMode`

## Possible solutions

The missing annotations should be listed and described.

## Additional information

n/a

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this pull request

This pull request (PR) updated the table with new annotation blocks.

Fixes #7936.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/test/integration/annotations.html

### CODE REVIEW

**Review by BarnyShergold — APPROVED** [review]

File: src/guides/v2.4/test/integration/annotations.md [hunk 1]
```diff
@@ -12,12 +12,15 @@ The following annotations are available in integration tests:
 
 Name|Annotation|Format|Description
 ---|---|---|---
+Admin Configuration Fixture|`@magentoAdminConfigFixture`|`@magentoAdminConfigFixture <config_path> <config_value>`| Configures an admin setting for the test run. For example, to enabled Captcha, you would use `@magentoAdminConfigFixture admin/captcha/enable 1`.
```
Comment (nathanjosiah) [inline src/guides/v2.4/test/integration/annotations.md:15]: `For example, to enabled Captcha,` => `For example, to enable Captcha,`

File: src/guides/v2.4/test/integration/annotations.md [hunk 2]
```diff
@@ -12,12 +12,15 @@ The following annotations are available in integration tests:
...
 Database Isolation|`@magentoDbIsolation`|`@magentoDbIsolation enabled|disabled`|Enables or disables database isolation. Disabled by default, unless you are using `@magentoDataFixture`, in which case it is enabled by default. All data, required for a test, live during transaction only. Any test results won't be written in a database.
-Data Fixture|`@magentoDataFixture`|`@magentoDataFixture <script_filename>|<method_name>`|Points to a class or a method which creates testing entities (fixtures) for test execution.
+Data Fixture|`@magentoDataFixture`|`@magentoDataFixture <script_filename>|<method_name>`|Points to a class or a method which creates testing entities (fixtures) for test execution. These are applied during the transaction.
+Data Fixture Before Transaction|`@magentoDataFixtureBeforeTransaction`|`@magentoDataFixtureBeforeTransaction <script_filename>|<method_name>`|Points to a class or a method which creates testing entities (fixtures) for test execution before the transaction has begun. You will need to implement a rollback for changes made here.
```
Comment (nathanjosiah) [inline src/guides/v2.4/test/integration/annotations.md:20]: Maybe worth an example for clarity? 
> You will need to implement a rollback file for changes made here. (e.g. Fixture file `my_fixture.php` would also require a `my_fixture_rollback.php` that reverts the original fixture's changed.)

File: src/guides/v2.4/test/integration/annotations.md [hunk 3]
```diff
@@ -12,12 +12,15 @@ The following annotations are available in integration tests:
...
 Application Area|`@magentoAppArea`|`@magentoAppArea adminhtml|frontend|global`|Configures test environment in the context of specified application area.
 Enable/Disable Cache|`@magentoCache`|`@magentoCache <type>|all enabled|disabled`|Enables or disables certain cache segment or all of them to prevent isolation problems.
+Indexer Dimension Mode|`@magentoIndexerDimensionMode`|`@magentoIndexerDimensionMode <indexer> <mode>`|Sets the indexer dimension mode for the test run. More information can be found in the [DevBlog](https://community.magento.com/t5/Magento-DevBlog/Indexers-parallelization-and-optimization/ba-p/104922_).
```
Comment (nathanjosiah) [inline src/guides/v2.4/test/integration/annotations.md:23]: I still feel like this doesn't make any sense as a random blog post. This blog post is technical documentation that can't be found anywhere else.

**Review by nathanjosiah — APPROVED** [review]

File: src/guides/v2.4/test/integration/annotations.md (hunk 3 above)
Comment (dobooth) [reply src/guides/v2.4/test/integration/annotations.md:23]: I agree. I am going to see if there is a place on devdocs for this.

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Review by shrielenee — APPROVED** [review]

**Pull request merged**